src/routes/contract.ts

```diff
@@ -84,38 +86,42 @@ export async function contractInteractionsHandler(ctx: KoaContext) {
       sortKey: requestedSortKey,
       blockHeight: requestedBlockHeight,
     }),
-    getWalletInteractionsForContract(arweave, {
+    getWalletInteractionsForContract({
+      arweave,
       address,
       contractTxId,
       blockHeight: requestedBlockHeight,
+      logger,
     }),
   ]);
 
+  logger.debug('Mapping interactions', {
+    contractTxId,
+    sortKey: requestedSortKey,
+    blockHeight: requestedBlockHeight,
+    address,
+  });
+
   let mappedInteractions = Array.from(interactions)
-    .map(
-      ([id, interaction]: [
-        string,
-        Omit<ArNSInteraction, 'valid' | 'errorMessage' | 'id'>,
-      ]) => {
-        // found in graphql but not by warp
-        if (!Object.keys(validity).includes(id)) {
-          logger.debug(
-            'Interaction found via GraphQL but not evaluated by warp',
-            {
-              contractTxId,
-              interaction: id,
-            },
-          );
-          mismatchedInteractionCount.inc();
-        }
-        return {
-          ...interaction,
-          valid: validity[id] ?? false,
-          error: errorMessages[id],
-          id,
-        };
-      },
-    )
+    .map(([id, interaction]: [string, GenericContractInteraction]) => {
```

Comment:
interesting to see this is taking so much time. we convert from a set -> array -> map over it -> sort.

Reply:
based on reviewing the logs i think the time between is due to the async operations (and pagination) in `readThroughToWalletInteractionsForContract` - we paginate through all the requests, then asynchronously assign sort keys to each iteraction retrieved. it doesn't appear to be an issue with the `map().sort()` we do here.